cgen: remove more obsolete code

zerbina opened the pull request:
## Summary

Remove leftover obsolete code from `cgen`:
* the `earlyResultAsgn` procedure
* the `nfPreventCg` node flag
* the `lfNoDeepCopy` loc flag
* the `TAssignmentFlag` enum and usages thereof

Thanks to ARC, there's no difference between `nkAsgn` and `nkFastAsgn`
in the C code generator, so the two branches are collapsed into a single
one.

saem reviewed (approved):
yay, code is collapsing down further!

saem commented:
/merge

zerbina committed:
cgen: remove more obsolete code

Summary
=======

Remove leftover obsolete code from `cgen`:
* the `earlyResultAsgn` procedure
* the `nfPreventCg` node flag
* the `lfNoDeepCopy` loc flag
* the `TAssignmentFlag` enum and usages thereof

Thanks to ARC, there's no difference between `nkAsgn` and `nkFastAsgn`
in the C code generator, so the two branches are collapsed into a single
one.
